dialog/fileitem.go

```diff
@@ -35,44 +41,67 @@ func (i *fileDialogItem) TappedSecondary(_ *fyne.PointEvent) {
 }
 
 func (i *fileDialogItem) CreateRenderer() fyne.WidgetRenderer {
-	img := canvas.NewImageFromResource(i.icon)
+	var img fyne.CanvasObject
+	if i.icon == nil {
+		mimeSub := i.mimeSubType
+		if len(mimeSub) > 5 {
```

Comment:
This doesn't seem to be the right place to trim a string - the trimming is a display detail and so should not pass in partial information